### DISCUSSION TITLE
inputstream adaptative/ffmpeg and qualities

### DISCUSSION THREAD

**Discussion by darodi** [Catch-up-TV-and-More/plugin.video.catchuptvandmore#692] [Ideas]
Following 

- #638 
- #385 
- #265
- #627

---

I notice:

1.  In #385, 
    the user doesn't want inputstream adaptative (conflicts), it should be not mandatory with a config switch
    I've also noticed inputstream adaptative is sometimes resource intensive and I would like to be able to disable it

2. in #265 

    > The other case is when we grab a m3u8 file. In this case our BEST|DEFAULT|DIALOG setting does nothing and we directly give this file too Kodi player.
    >So Kodi will select the best quality depending on your network. Also you can manually select the bitrate that you want with the select program Kodi setting.
    
    but you cannot force or choose a lower setting before play and this is wanted in #385 

3. in 
   #638 , 
   #627
   we would like to be able to force the quality or choose from bandwidth

---

Some other add-ons had the same problems, they implemented some elegant solutions. 
Here for example:
https://www.matthuisman.nz/2019/09/new-quality-select-in-my-add-ons.html

A list of interesting features:

- [ ] The available qualities options
- [ ]  Context Menu before play
- [ ]  Ability to choose inputstream for HLS or not
- [ ]  It first downloads the playlist file and grabs the available bandwidths.
- [ ]  For Inputstream, this is the maximum and minimum bandwidth.
- [ ]  For ffmpeg, it sets the Kodi Internet Bandwidth limitation setting.

I think we should not reinvent the wheel and rely on this idea.
What are your thoughts?

**Comment by joaopa00** [comment]
Do you want to merge parts of Huisman's addon in catchuptvandmore? It looks a huge task.

**Comment by joaopa00** [comment]
Je pense que l'on devrait essayer de modifier tous les fichiers (là où c'est possible) pour utiliser le schéma. de la PR #706 Il faudrait  ajouter une option dans les paramètres de Catchuptvandmore pour proposer comme choix

choix 
1 - (par défaut) utilser inputstream adaptive
2 - dialogue 
3  -Si Inputstream n'est pas présent, vidéo par défaut (sans dialogue)

Il ne faut pas embêter l'utilisateur qui veut une configuration par défaut et cliquer sur sa chaîne sans rien faire d'autre.

Quand cela sera activé sur Inputstream adaptive (c'est déjà implémenter voir [github Inputstream](https://github.com/xbmc/inputstream.adaptive) TODO), la résolution s"ajustera directement sur la bande passante disponible. Ce sera le gros plus de Inputstream adaptive. Je ne pense pas qu'il soit utile d'implémenter le choix de la qualité à travers Inputstream.

**Comment by joaopa00** [comment]
Je pense que le projet est déjà bien avancé. Autant que faire se peut, j'essaie à chaque modification de l'utiliser. Je pense que l'on peut clore cette dscussion comme achevée.
